module/x/gravity/keeper/evidence.go

```diff
@@ -0,0 +1,93 @@
+package keeper
+
+import (
+	"bytes"
+	"encoding/hex"
+	"fmt"
+
+	sdk "github.com/cosmos/cosmos-sdk/types"
+	sdkerrors "github.com/cosmos/cosmos-sdk/types/errors"
+
+	"github.com/cosmos/gravity-bridge/module/x/gravity/types"
+)
+
+func (k Keeper) CheckBadSignatureEvidence(
+	ctx sdk.Context,
+	msg *types.MsgSubmitBadSignatureEvidence) error {
```

Comment:
should be misbehavior instead of the full msg

Reply:
Are you referring to the naming or something else?